src/ansi-c/library/jsa.h

```diff
@@ -188,7 +188,10 @@ __CPROVER_jsa_inline _Bool __CPROVER_jsa__internal_are_heaps_equal(
   const __CPROVER_jsa_abstract_heapt *const rhs)
 {
   __CPROVER_jsa__internal_index_t i;
+#include <util/pragma_push.def>
+#include <util/pragma_wtautological_compare.def>
   for(i=0; i < __CPROVER_JSA_MAX_ABSTRACT_NODES; ++i)
+#include <util/pragma_pop.def>
```

Comment:
Having had another look at the code I do now seem to understand. For all cases in this file, could you please wrap the `for` loop (and also the preceding declaration by `#if __CPROVER_JSA_MAX_ABSTRACT_NODES > 0` - this will resolve the need for the pragma in those loops. (This is all due to cegis/jsa/value/jsa_genetic_synthesis.h.)